packages/@uppy/aws-s3-multipart/types/index.d.ts

```diff
@@ -25,6 +35,7 @@ export interface AwsS3MultipartOptions extends PluginOptions {
       file: UppyFile,
       opts: { uploadId: string; key: string; signal: AbortSignal }
     ) => MaybePromise<AwsS3Part[]>
+    getTemporarySecurityCredentials?: boolean | (() => MaybePromise<AwsS3STSResponse>)
```

Comment:
I think it's fair to assume you always need a promise, or that we simply expect it.
```suggestion
    getTemporarySecurityCredentials?: boolean | (() => Promise<AwsS3STSResponse>)
```

Reply:
But why?